psa-crypto/src/types/algorithm.rs

```diff
@@ -578,10 +579,10 @@ impl TryFrom<psa_crypto_sys::psa_algorithm_t> for Algorithm {
 impl TryFrom<Algorithm> for psa_crypto_sys::psa_algorithm_t {
     type Error = Error;
     fn try_from(alg: Algorithm) -> Result<Self> {
-        match alg {
+        match &alg {
```

Comment:
Why is it necessary to match on the reference here and clone later instead of matching the value and passing ownership?

Reply:
Because for a type that implements `Drop` you can't take any elements out (otherwise you might not have everything you need in the object once you get into `drop`)